dashboard/lib/pd/jot_form/translation.rb

```diff
@@ -167,11 +167,12 @@ def parse_jotform_submission(jotform_submission)
       # For each non-datetime hash, also remove "\\" pairs from each string as
       # well, which are appearing in strings such as "I\\'m a teacher."
       def strip_answer(answer)
-        if answer.is_a? String
+        case answer
+        when String
           answer.strip
-        elsif answer.is_a? Array
+        when Array
           answer.map(&:strip)
-        elsif answer.is_a? Hash
+        when Hash
```

Comment:
Does `case` automatically treat this as an `is_a` check vs an equality check?

Reply:
Yep! Ruby has a special "case equality operator" (`===`) that it uses for `case` statements and which is a bit weird but generally works as a "is the second argument a member of the set defined by the first argument". See also the regular expressions [here](https://github.com/code-dot-org/code-dot-org/pull/50054/files#diff-51fff2d994ef31ee8e7d74fdf41dcc9a2654c84e154e1bf55be47e01b346b4f7L43-R47); the case equality operator automatically checks for a match rather than an equality check

See https://stackoverflow.com/a/4467823/1810460 and https://blog.arkency.com/the-equals-equals-equals-case-equality-operator-in-ruby/ for more

Reply:
Neat!